### PR TITLE
Use referrer to determine URL sent in emails

### LINKED ISSUE #112: Reset password comes as a Mandrillapp tracking link with wrong path
**- Do you want to request a *feature* or report a *bug*?**
Bug

**- What is the current behavior?**
Reset password link is a Mandrill link, which in turn makes the reset password unsuccessful.
After tracking the link, the redirected path is wrong: example.com/#recovery_token=somenumbers
missing **_admin_** from the path.

**- If the current behavior is a bug, please provide the steps to reproduce.**
1.- create a netlifycms instance
2.- change to a theme that uses the simple Netlify Identity login (not the git identity)
3.- Create a user inside app.netlify.com
4.- Send password reset 

**- What is the expected behavior?**
Link is a reset password

**- Please mention your Go version, and operating system version.**
OSX Sierra, 
Go version unknown (command not found)

### PR DESCRIPTION
**Pull request by brycekahle**
**- Summary**
Fixes #112

**- Test plan**
Added tests for email URL generation function w/ referrer included.

**- Description for the changelog**
Use referrer to determine URL sent in emails.

### CODE REVIEW

File: api/helpers.go [hunk 1]
```diff
@@ -91,3 +92,18 @@ func (a *API) requestAud(ctx context.Context, r *http.Request) string {
...
 }
+
+func (a *API) getReferrer(r *http.Request) string {
+	ctx := r.Context()
+	config := a.getConfig(ctx)
+	referrer := ""
+	if reqref := r.Referer(); reqref != "" {
+		base, berr := url.Parse(config.SiteURL)
+		refurl, rerr := url.Parse(reqref)
+		// As long as the referrer came from the site, we will redirect back there
```
Comment (bcomnes) [inline api/helpers.go]: What about localhost?

Comment (brycekahle) [reply api/helpers.go]: What do you mean?

Comment (bcomnes) [reply api/helpers.go]: If you try to log in from a site running on localhost, will the confirm emails pick that up in the redirect they send you?

Comment (brycekahle) [reply api/helpers.go]: Only if the site URL is also localhost

**Pull request merged**